jbpm-persistence/jbpm-persistence-jpa/src/test/java/org/jbpm/persistence/correlation/CorrelationPersistenceTest.java

```diff
@@ -70,13 +70,18 @@ public void before() throws Exception {
         ut.begin();
         EntityManager em = emf.createEntityManager();
 
-        em.persist(factory.newCorrelationKey("test123"));
+        try {
+            em.persist(factory.newCorrelationKey("test123"));
 
-        List<String> props = new ArrayList<String>();
-        props.add("test123");
-        props.add("123test");
-        em.persist(factory.newCorrelationKey(props));
-        
+            List<String> props = new ArrayList<String>();
+            props.add("test123");
+            props.add("123test");
+
+            em.persist(factory.newCorrelationKey(props));
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            ut.rollback();
```

Comment:
In case there is an exception, wouldn't this let the test pass?  shouldn't we add a fail here?

Reply:
It should not, because there is  ut.commit() right after the try-catch block, which will fail as there is no active transaction due to ut.rollback() call in catch block. But, I agree that Assert.fail(); is better option here.

Reply:
@domhanak could you please also rather use logging instead of ex.printStackTrace() ?

Reply:
I think the whole transaction should be in try-catch block. See here: https://gist.github.com/baldimir/334084038b994ceb25c5b09d2cedb1ec

When anything fails after transaction began, it must be rolled back.